File: website/lib/contracts/external/pinned_repositories_file.ts
export default interface PinnedRepositoriesFile {
    data: Array<GitHubRepository>;
}

export interface GitHubRepository {
    name: string;
    description: string;
    url: string;
    languages: Array<GitHubLanguage>;
}

export interface GitHubLanguage {
    name: string
    color: string
}

File: serverless/GithubSyncer/GithubSyncer/Contracts/External/S3/PinnedRepositoriesFile.cs
using GithubSyncer.Contracts.External.Github;

namespace GithubSyncer.Contracts.External.S3;

public class PinnedRepositoriesFile
{
    public IEnumerable<GithubRepository> Data { get; set; } = Enumerable.Empty<GithubRepository>();

    public class GithubRepository
    {
        public string Name { get; set; } = string.Empty;
        public string Url { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public IEnumerable<GithubLanguage> Languages { get; set; } = Enumerable.Empty<GithubLanguage>();
    }
}
